### PR TITLE
chore: allow specific token sale in secondary market

### PR DESCRIPTION
**Pull request by Peartes**
This PR adds in functionality matching latest secondary market specs

Major change from previous purchase logic is being able to by a specific token as to the lowest listed token.

### CODE REVIEW

File: sellable/src/execute.rs [hunk 1]
```diff
@@ -58,7 +59,75 @@ where
...
+    pub fn try_buy_token(
+        &mut self,
+        deps: &mut DepsMut,
+        info: MessageInfo,
+        token_id: String,
+    ) -> Result<Response, ContractError> {
+        // check if enough fee was sent
+        if info.funds.len() == 0 {
+            return Err(ContractError::NoFundsPresent);
+        } else if info.funds.len() > 1 {
```
Comment (bigs) [inline sellable/src/execute.rs:71]: We can make this slightly simpler for ourselves using pattern matching

```rust
match info.funds.as_slice() {
  [coin] => { /* this is the branch in which there is a single item */ }
  [] => { /* this is the branch where there are no items */ }
  _ => { /* this is the branch where there are multiple items */ }
}
```

The advantage of this approach is that we've already destructured our vector in the happy path and can directly address `coin`

File: sellable/src/execute.rs [hunk 2]
```diff
@@ -58,7 +59,75 @@ where
...
+            return Err(ContractError::MultipleFundsError);
+        } else {
+            let token = self
+                .listed_tokens
+                .load(deps.storage, token_id.as_str())
+                .map_err(|_| ContractError::NoListedTokensError);
+            match token {
```
Comment (bigs) [inline sellable/src/execute.rs:78]: This match can be replaced with a flat_map e.g.

```rust
self
  .listed_tokens
  .load(deps.storage, token_id.as_str())
  .map_err(|_| ContractError::NoListedTokensError)
  .flat_map(|price| {
    // normal logic here where you can return Err or Ok
  })

File: sellable/src/execute.rs (hunk 1 above)
Comment (Peartes) [reply sellable/src/execute.rs:71]: this is lovely logic

File: sellable/src/execute.rs (hunk 2 above)
Comment (Peartes) [reply sellable/src/execute.rs:78]: the return type of the load is a Result<Coin, Error> 
unlike it name "listed_tokens", it only loads a single token

**Pull request merged**